src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,459 @@
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                if (!IsWithinConditionalOperation(operation))
+                    attributes = FindAllPlatformAttributesApplied(pOperation.Property.GetAttributes(), pOperation.Property.ContainingType);
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                if (!IsWithinConditionalOperation(operation))
+                    attributes = FindAllPlatformAttributesApplied(fOperation.Field.GetAttributes(), fOperation.Field.ContainingType);
+            }
+
+            if (attributes != null)
+            {
+                foreach (AttributeData attribute in attributes)
+                {
+                    if (PlatformAttributeInfo.TryParseAttributeData(attribute, out PlatformAttributeInfo parsedAttribute))
+                    {
+                        if (!(IsSuppressedByTfm(parsedTfms, parsedAttribute) || IsSuppressedByAttribute(parsedAttribute, context.ContainingSymbol)))
+                        {
+                            builder.Add(parsedAttribute);
+                        }
+                    }
+                }
+            }
+
+            if (builder.Count > 0)
+            {
+                platformSpecificOperations.Add(operation, builder.ToImmutable());
+            }
+        }
+
+        private static bool IsWithinConditionalOperation(IOperation pOperation) =>
+         pOperation.Parent is IBinaryOperation bo &&
+            (bo.OperatorKind == BinaryOperatorKind.Equals ||
+            bo.OperatorKind == BinaryOperatorKind.NotEquals ||
+            bo.OperatorKind == BinaryOperatorKind.GreaterThan ||
+            bo.OperatorKind == BinaryOperatorKind.LessThan ||
+            bo.OperatorKind == BinaryOperatorKind.GreaterThanOrEqual ||
+            bo.OperatorKind == BinaryOperatorKind.LessThanOrEqual);
+
+        private static bool IsSuppressedByTfm(List<PlatformAttributeInfo>? parsedTfms, PlatformAttributeInfo parsedAttribute)
+        {
+            if (parsedTfms != null)
+            {
+                foreach (PlatformAttributeInfo tfm in parsedTfms)
+                {
+                    if (IsOSPlatformsEqual(parsedAttribute.OsPlatformName, tfm.OsPlatformName))
+                    {
+                        return AttributeVersionsMatch(parsedAttribute, tfm);
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private static bool IsOSPlatformsEqual(string firstOs, string secondOs)
+        {
+            return firstOs.Equals(secondOs, StringComparison.OrdinalIgnoreCase)
+                || (firstOs.Equals("OSX", StringComparison.OrdinalIgnoreCase) && secondOs.Equals("MACOS", StringComparison.OrdinalIgnoreCase))
+                || (secondOs.Equals("OSX", StringComparison.OrdinalIgnoreCase) && firstOs.Equals("MACOS", StringComparison.OrdinalIgnoreCase));
+        }
```

Comment:
Might be more readable to do it like this:

```suggestion
        private static string NormalizeOSName(string name)
        {
            if (name.Equals("OSX", StringComparison.OrdinalIgnoreCase))
                return "macOS";

            return name;
        }

        private static bool IsOSPlatformsEqual(string left, string right)
        {
            var normalizedLeft = NormalizeOSName(left);
            var normalizedRight = NormalizeOSName(right);
            return string.Equals(normalizedLeft, normalizedRight, StringComparison.OrdinalIgnoreCase);
        }
```